pkg/controllers/uiplugin/components.go

```diff
@@ -19,21 +23,59 @@ const (
 	port                  = 9443
 	serviceAccountSuffix  = "-sa"
 	servingCertVolumeName = "serving-cert"
+	Korrel8rConfigFileName = "korrel8r.yaml"
+	Korrel8rConfigMountDir = "/config/"
 )
 
+var (
+		//go:embed config/korrel8r.yaml
+		korrel8rConfigYAMLTmplFile embed.FS
+)
+
+
 func pluginComponentReconcilers(plugin *uiv1alpha1.UIPlugin, pluginInfo UIPluginInfo) []reconciler.Reconciler {
-	hasRole := pluginInfo.Role != nil
-	hasRoleBinding := pluginInfo.RoleBinding != nil
 	namespace := pluginInfo.ResourceNamespace
+	kname := "korrel8r"
 
-	return []reconciler.Reconciler{
+	components := []reconciler.Reconciler{
 		reconciler.NewUpdater(newServiceAccount(pluginInfo, namespace), plugin),
-		reconciler.NewOptionalUpdater(newRole(pluginInfo), plugin, hasRole),
-		reconciler.NewOptionalUpdater(newRoleBinding(pluginInfo), plugin, hasRoleBinding),
 		reconciler.NewUpdater(newDeployment(pluginInfo, namespace), plugin),
 		reconciler.NewUpdater(newService(pluginInfo, namespace), plugin),
 		reconciler.NewUpdater(newConsolePlugin(pluginInfo, namespace), plugin),
+		reconciler.NewUpdater(newKorrel8rService(kname, namespace), plugin),
```

Comment:
I made the deployment of Korrel8r components conditional. But I thought the components creation can still be in components.go file., since all the resource creation seems to be in there.

Reply:
@jgbernalp does this mean that the existing korrel8r links (alerts to logs, logs to metrics) are being dropped, or that they're only available if the troubleshooting panel is present? I've no objection either way, just want to know.

Reply:
the logging plugin is the one enabling the links if korrel8r is found in the cluster. In this case, the troubleshooting panel will be the trigger to enable korrel8r. So yes, they will be enabled only when the troubleshooting panel is present.

Reply:
+1